app/controllers/application_controller.rb

```diff
@@ -12,11 +12,11 @@ class ApplicationController < ActionController::Base
 
   _helpers.module_eval { prepend DecoratorAdditions }
 
-  protect_from_forgery with: :reset_session # See ActionController::RequestForgeryProtection for details
+  protect_from_forgery with: ActionController::RequestForgeryProtection::ExceptionAndResetStrategy
 
   # Disable CSRF protection for requests to REST API.
   skip_before_action :verify_authenticity_token, if: -> do
-    api_controller? && (params.key?(:provider_key) || params.key?(:access_token))
+    api_controller? && logged_in?
```

Comment:
This fixes the `basic` auth.

Before this, any request including `provider_key` or `access_token` skipped the forgery protection, no matter whether the token was right or not. It also forced forgery protection for API requests including the tokens by other means like the `basic` auth.

Reply:
maybe make it one-line? Just a nitpick